bug: Windows MSVC release build fails with unresolved libc symbols

## Summary

The Release Binaries workflow fails for `x86_64-pc-windows-msvc` with linker errors for four libc functions.

## Error

From [run #22211846804](https://github.com/wowemulation-dev/rilua/actions/runs/22211846804):

```
error LNK2019: unresolved external symbol gmtime_s referenced in function _ZN5rilua6stdlib2os7os_date...
error LNK2019: unresolved external symbol localtime_s referenced in function _ZN5rilua6stdlib2os7os_date...
error LNK2019: unresolved external symbol fprintf referenced in function _ZN5rilua6stdlib2io7g_write...
error LNK2019: unresolved external symbol fscanf referenced in function _ZN5rilua6stdlib2io6g_read...
fatal error LNK1120: 4 unresolved externals
```

## Affected symbols

| Symbol | File | Function |
|--------|------|----------|
| `gmtime_s` | `src/stdlib/os.rs` | `os_date` (UTC) |
| `localtime_s` | `src/stdlib/os.rs` | `os_date` (local) |
| `fprintf` | `src/stdlib/io.rs` | `g_write` |
| `fscanf` | `src/stdlib/io.rs` | `g_read` |

## Cause

The FFI declarations in `src/platform.rs` reference C runtime functions that require explicit linking on Windows MSVC. The CI test workflow passes because it runs `cargo test` on Linux, but the release binary workflow cross-compiles with `--target x86_64-pc-windows-msvc` where the universal C runtime (`ucrt`) is not linked automatically.

## Possible fixes

1. Add `#[link(name = "ucrt")]` or `#[link(name = "msvcrt")]` to the FFI extern blocks on Windows
2. Use `#[cfg_attr(target_env = "msvc", link(name = "ucrt"))]` conditionally
3. Add a `build.rs` that emits `cargo:rustc-link-lib=ucrt` on MSVC targets

## First observed

Tag v0.1.8, workflow run on 2026-02-20.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,10 @@ and this project adheres to
 
 ## [Unreleased]
 
+### Fixed
+
+- Link `ucrt` on Windows MSVC for libc FFI symbols (`gmtime_s`, `localtime_s`, `fprintf`, `fscanf`)
+
 ## [0.1.9](https://github.com/wowemulation-dev/rilua/compare/v0.1.8...v0.1.9) - 2026-02-20
 
 ### Added

--- a/src/platform.rs
+++ b/src/platform.rs
@@ -28,6 +28,7 @@ pub(crate) enum LibcFile {}
 // ---------------------------------------------------------------------------
 
 #[allow(unsafe_code)]
+#[cfg_attr(target_env = "msvc", link(name = "ucrt"))]
 unsafe extern "C" {
     pub(crate) fn fopen(filename: *const u8, mode: *const u8) -> *mut LibcFile;
     pub(crate) fn fclose(file: *mut LibcFile) -> i32;
@@ -190,6 +191,7 @@ unsafe extern "C" {
 
 #[cfg(target_os = "windows")]
 #[allow(unsafe_code)]
+#[link(name = "ucrt")]
 unsafe extern "C" {
     fn __acrt_iob_func(index: u32) -> *mut LibcFile;
 }
@@ -252,6 +254,7 @@ unsafe extern "C" {
 
 #[cfg(target_os = "windows")]
 #[allow(unsafe_code)]
+#[link(name = "ucrt")]
 unsafe extern "C" {
     #[link_name = "_popen"]
     fn popen(command: *const u8, r#type: *const u8) -> *mut LibcFile;
@@ -284,6 +287,7 @@ unsafe extern "C" {
 
 #[cfg(target_os = "windows")]
 #[allow(unsafe_code)]
+#[link(name = "ucrt")]
 unsafe extern "C" {
     // Windows reverses parameter order and returns errno_t.
     fn localtime_s(result: *mut Tm, timep: *const TimeT) -> i32;

